fp-core/src/functor.rs

```diff
@@ -1,15 +1,15 @@
 use crate::hkt::HKT;
 
-pub trait Functor<A, B>: HKT<A, B> {
-    fn fmap<F>(self, f: F) -> <Self as HKT<A, B>>::Target
+pub trait Functor<B>: HKT<B> {
+    fn fmap<F>(self, f: F) -> <Self as HKT<B>>::Target
     where
-        F: FnOnce(A) -> B;
+        F: FnOnce(<Self as HKT<B>>::Current) -> B;
 }
 
-impl<A, B> Functor<A, B> for Option<A> {
+impl<A, B> Functor<B> for Option<A> {
     fn fmap<F>(self, f: F) -> Self::Target
```

Comment:
I feel like we can now name this `map`

Reply:
We'd likely conflict all over the place with `Iterator`'s map.

Reply:
I agree on this, thanks =)